tests/system/small/test_unordered.py

```diff
@@ -59,3 +61,51 @@ def test_unordered_mode_read_gbq(unordered_session):
     )
     # Don't need ignore_order as there is only 1 row
     assert_pandas_df_equal(df.to_pandas(), expected)
+
+
+@pytest.mark.parametrize(
+    ("keep"),
+    [
+        pytest.param(
+            "first",
+            marks=pytest.mark.xfail(raises=bigframes.exceptions.OrderRequiredError),
+        ),
+        pytest.param(
+            False,
+        ),
+    ],
+)
+def test_unordered_drop_duplicates(unordered_session, keep):
+    pd_df = pd.DataFrame({"a": [1, 1, 3], "b": [4, 4, 6]}, dtype=pd.Int64Dtype())
+    bf_df = bpd.DataFrame(pd_df, session=unordered_session)
+
+    bf_result = bf_df.drop_duplicates(keep=keep)
+    pd_result = pd_df.drop_duplicates(keep=keep)
+
+    assert_pandas_df_equal(bf_result.to_pandas(), pd_result, ignore_order=True)
+
+
+@pytest.mark.parametrize(
+    ("function"),
+    [
+        pytest.param(
+            lambda x: x.cumsum(),
+            id="cumsum",
+            marks=pytest.mark.xfail(raises=bigframes.exceptions.OrderRequiredError),
```

Comment:
Since all of these tests fail with the same exception, prefer `with pytest.raises(...)` in the test function body. Ideally we check for the function name in the error message with `match=` as well.

Reply:
done